src/App.vue

```diff
@@ -1,62 +1,128 @@
+
+                <b-collapse id="dataPreview" class="">
+                  <vue-json-pretty :data="previewData" class="card-body"></vue-json-pretty>
+                </b-collapse>
+              </div>
+
+              <div id="data-input" class="overflow-auto">
+                  <b-button v-b-toggle.dataInput variant="outline-*" class="text-left card-header d-flex align-items-center sticky-top header-bg w-100" @click="showDataInput = !showDataInput">
+                    <i class="fas fa-file-import mr-2"></i>
+                      Data Input
+                    <i class="fas fa-angle-down ml-auto" :class="{ 'fas fa-angle-right' : !showDataInput }"></i>
+                  </b-button>
+
+                  <b-collapse id="dataInput">
+                      <form-text-area class="dataInput" rows="8" v-model="previewInput"></form-text-area>
+                  </b-collapse>
+              </div>
             </div>
+          </div>
+        </div>
 
-            <div id="renderer-container" class="w-50 p-4 pt-5 overflow-auto mb-5">
-                <div class="container">
-                    <div class="row">
-                        <div class="col-sm">
-                            <vue-form-renderer ref="renderer"
-                                               v-model="previewData"
-                                               @submit="previewSubmit"
-                                               :config="config"
-                                               :computed="computed"
-                                               :custom-css="customCSS"
-                                               v-on:css-errors="cssErrors = $event"/>
-                        </div>
-                    </div>
-                </div>
+        <div class="card-footer text-muted d-flex justify-content-end align-items-center">
+            <div>
+              <span class="custom-control custom-switch">
+              <input v-model="toggleValidation" type="checkbox" class="custom-control-input" id="customSwitch1" checked>
+              <label class="custom-control-label" for="customSwitch1"> Screen Validation
+              </label>
+              </span>
             </div>
 
-            <div id="data-preview" class="w-25 border overflow-auto mb-5">
-                <div class="card-header">
-                    {{$t('Data Preview')}}
+            <div v-if="showValidationErrors" class="validation-panel position-absolute shadow border overflow-auto" :class="{'d-block':showValidationErrors && validationErrors.length}">
+                <div v-if="!previewInputValid" class="p-3 font-weight-bold text-dark">
+                  <i class="fas fa-times-circle text-danger mr-3"></i>
+                  {{$t('Invalid JSON Data Object')}}
                 </div>
-                <vue-json-pretty :data="previewData" class="card-body"></vue-json-pretty>
+                <a v-for="(validation,index) in validationErrors" :key="index" href="javascript:void(0)"
```

Comment:
I'm not a fan of using `href="javascript:void(0)"` to disguise a link as a button. I think it might be better to use `<b-button variant="link">Link</b-button>`.

Reply:
I agree